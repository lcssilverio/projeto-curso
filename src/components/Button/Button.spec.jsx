import {fireEvent, render, screen} from '@testing-library/react'
import {Button} from '.'

describe('Tests to Button component', () => {
  test('should render the button with the text', () => {
    render(<Button text="Load more"/>)
    expect.assertions(1)

    const button = screen.getByRole('button', {name: /load more/i })
    expect(button).toBeInTheDocument()
  })
  test('should call function on button click', () => {
    const handleClick = jest.fn()
    render(<Button text="Load more" onClick={handleClick}/>)
    const button = screen.getByRole('button', {name: /load more/i })
    fireEvent.click(button)

    expect(handleClick).toHaveBeenCalled()
  })
  test('should be disabled when disabled is true', () => {
    render(<Button text="Load more" disabled={true}/>)

    const button = screen.getByRole('button', {name: /load more/i })
    expect(button).toBeDisabled()
  })

  test('should be enable when enable is true', () => {
    render(<Button text="Load more" disabled={false}/>)

    const button = screen.getByRole('button', {name: /load more/i })
    expect(button).not.toBeDisabled()
  })

  test('should match  snapshot', () => {
    const handleClick = jest.fn()
    const { container } = render(<Button text="Load more" onClick={handleClick}/>)

    expect(container).toMatchSnapshot()
  })
})
