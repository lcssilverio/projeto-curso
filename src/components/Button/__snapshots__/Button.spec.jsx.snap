// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests to Button component should match  snapshot 1`] = `
<div>
  <button
    class="button"
  >
    Load more
  </button>
</div>
`;
